### PR TITLE
Console Calculator, challenges done

### PR DESCRIPTION
**Pull request by PiotrrKubas**

### CODE REVIEW

File: CalculatorLibrary/CalculatorLibrary.cs [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                    break;
+                case "*":
+                    result = num1 * num2;
+                    writer.WriteValue("Multiply");
+                    break;           
+                case "/":
+                    if (num2 != 0)
+                    {
+                        result = num1 / num2;
+                        writer.WriteValue("Divide");
```
Comment (chrisjamiecarter) [inline CalculatorLibrary/CalculatorLibrary.cs:132]: As per the tutorial this is incorrect and will result in the wrong error 😄 .

Test and see for yourself 😱 !

After you have tested dividing by zero please update as per the microsoft documentation:
https://learn.microsoft.com/en-us/visualstudio/get-started/csharp/tutorial-console-part-2?view=vs-2022

**Review by chrisjamiecarter — CHANGES_REQUESTED** [review]
Hey @PiotrrKubas 👋,

First of all, great job on your calculator application! 🎉 It’s clear you’ve put a lot of effort into this, and you’re making great progress.
I realize that much of your code likely follows the tutorial, but 
I have performed a peer review and have some praise, as well as a few suggestions:
- Great job on completing the challenges ⭐ 


**Solution Structure**
Re-read step 3 of the [Code Reviews](https://thecsharpacademy.com/article/52/code-reviews) academy page.
This will make it easier for code-reviewers to pull down and review your code in the future.
When submitting your code review pull request, all your solution files should be contained inside a new folder as per below:

✔️ 
```
.
└── [ProjectName].[GitHubUserName]/
    ├── [SolutionDirectory1]/
    ├── [SolutionDirectory2]/
    └── [ProjectName].sln
```

**Style Guide Compliance**
Remember that when you have finished a project, give it a polish pass. Pay attention to any code conventions in place, in this instance refer to the C# Academy's code-conventions page.
This is important as good code conventions/styling ensures consistency, makes your code easier for others to read and maintain, and improves your chances of getting a pull request reviewed and approved.

**Issues**
Unfortunately, I did spot an issue within your application.
Remember to test your application before submission. Attempt a Divide by Zero use case and see the results!

I suggest revisiting to understand and fix the issues, the commit and push your changes once resolved. This will notify me and I will (hopefully) approve this pull request.

Feel free to reach out if you have any questions or if you'd like to collaborate further on any of these points.

Thanks,
@chrisjamiecarter

**Comment by TheCSharpAcademy** [comment]
@PiotrrKubas Let us know if you need help 👌🏻

**Review by chrisjamiecarter — CHANGES_REQUESTED** [review]
Hey @PiotrrKubas 👋 

Great job on the updates 🎉 .

So close to the final solution 🐎 !

However, unfortunately, there is still a slight issue ☹️ .

Again, I tried the test the divide by 0 use case ➗ .
The result will always be 8. Which I think is just how my terminal wants to display infinity ♾️ .

image

The logic in your update isn't quite right 😵‍ .

```
case "/":
    bool isNum2Correct = double.IsInfinity(num1 / num2); // This should be: bool isNum2Correct = !double.IsInfinity(num1 / num2);
    while (isNum2Correct) // This should be: while (!isNum2Correct)
    {
        Console.Clear();
        Console.WriteLine("You cant divide by '0'");
        Console.WriteLine("Enter another divisor:");
        isNum2Correct = !double.TryParse(Console.ReadLine(), out num2);
    }
    result = num1 / num2;
    writer.WriteValue("Division");
    break;
```

**Or you could have done what I was expecting...** and move the line `writer.WriteValue("Division")` as per the tutorial 🤓 :

Your initial code:
image

Expected:
image

😄 
Hopefully this sheds some clearer light on the issue, and you can address and fix quickly!

Thanks,
@chrisjamiecarter

**Review by chrisjamiecarter — APPROVED** [review]
Perfect @PiotrrKubas 🎉 ,

I will go ahead and mark as approved, keep up the excellent work on the next projects! 😊

Best regards,
@chrisjamiecarter

**Pull request closed without merging**